src/platform_linux/platform.h

```diff
@@ -278,28 +293,67 @@ typedef uint32 (*hash_fn)(const void *input, size_t length, unsigned int seed);
  * DO NOT USE these macros to assign to a void*.  The debug asserts will cause
  * a compile error when debug is on.  Assigning to a void* should be done by
  * calling aligned_alloc manually (or create a separate macro)
+ *
+ * Parameters:
+ *	id  - Platform heap-ID to allocate memory from.
+ *	v   - Structure to allocate memory for.
+ *	n   - Number of bytes of memory to allocate.
+ * -----------------------------------------------------------------------------
  */
-#define TYPED_MALLOC_MANUAL(id, v, n)                                          \
+#define TYPED_MANUAL_MALLOC(id, v, n)                                          \
    ({                                                                          \
       debug_assert((n) >= sizeof(*(v)));                                       \
       (typeof(v))platform_aligned_malloc(id, PLATFORM_CACHELINE_SIZE, (n));    \
    })
-#define TYPED_ZALLOC_MANUAL(id, v, n)                                          \
+#define TYPED_MANUAL_ZALLOC(id, v, n)                                          \
    ({                                                                          \
       debug_assert((n) >= sizeof(*(v)));                                       \
       (typeof(v))platform_aligned_zalloc(id, PLATFORM_CACHELINE_SIZE, (n));    \
    })
 
 /*
+ * TYPED_ALIGNED_MALLOC(), TYPED_ALIGNED_ZALLOC()
+ *
+ * Allocate memory for a typed structure at caller-specified alignment.
+ * These are similar to TYPED_MANUAL_MALLOC() & TYPED_MANUAL_ZALLOC() but with
+ * the difference that the alignment is caller-specified.
+ *
+ * Parameters:
+ *	id  - Platform heap-ID to allocate memory from.
+ *	a   - Alignment needed for allocated memory.
+ *	v   - Structure to allocate memory for.
+ *	n   - Number of bytes of memory to allocate.
+ */
+#define TYPED_ALIGNED_MALLOC(id, a, v, n)                                      \
+   ({                                                                          \
+      debug_assert((n) >= sizeof(*(v)));                                       \
+      (typeof(v))platform_aligned_malloc(id, (a), (n));                        \
+   })
+#define TYPED_ALIGNED_ZALLOC(id, a, v, n)                                      \
```

Comment:
Added this pair of variants where the 2nd one is now used to replace inline code in `rc_allocator.c`.